src/lib/storage/table.hpp

```diff
@@ -79,17 +79,25 @@ class Table : private Noncopyable {
   // creates a new chunk and appends it
   void create_new_chunk();
 
+  // compresses a ValueColumn into a DictionaryColumn
+  void compress_chunk(ChunkID chunk_id);
+
  protected:
-  std::vector<bool> m_is_instantiated;
-  std::vector<std::string> m_column_names;
-  std::vector<std::string> m_column_types;
-  std::vector<Chunk> m_chunks;
-  uint32_t m_chunk_size;
+  // mark if a column was only defined (add_column_definition) or already instantiated (add_column)
+  std::vector<bool> _is_instantiated;
+  std::vector<std::string> _column_names;
+  std::vector<std::string> _column_types;
+  std::vector<Chunk> _chunks;
+  const uint32_t _chunk_size;
 
- private:
   bool is_last_chunk_full() const;
```

Comment:
@ teaching assistants: We forgot to check the contributor guidelines here again, sorry. Will add underscores before merging back to master.